File: app/src/main/java/dev/chu/architecture/java/p80/component/Cafe.java
//package dev.chu.architecture.java.p80.component;
//
//import javax.inject.Inject;
//
//import dev.chu.architecture.java.p80.sub_component.Coffee;
//import dev.chu.architecture.java.p80.sub_component.Machine;
//
//public class Cafe {
//
//    @Inject
//    Machine coffeeMachine;
//
//    public Cafe() {
//        DaggerCafeComponent.create().inject(this);
//    }
//
//    public Coffee orderCoffee() {
//        return coffeeMachine.extract();
//    }
//}

File: app/src/main/java/dev/chu/architecture/kotlin/p80/component/Cafe.kt
package dev.chu.architecture.kotlin.p80.component

import dev.chu.architecture.kotlin.p80.sub_component.Machine
import javax.inject.Inject

class Cafe @Inject constructor(){
    init {
        println("Cafe")
        DaggerCafeComponent.create().inject(this)
    }

    @Inject
    @JvmField
    var coffeeMachine: Machine? = null

    fun orderCoffee() = coffeeMachine?.extract()
}